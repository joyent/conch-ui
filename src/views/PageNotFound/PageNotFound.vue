<template>
  <div class="page-not-found hero is-fullheight">
    <div class="content-left">
      <h1 class="title">404</h1>
      <h2 class="subtitle is-5">
        The page you're looking for is... missing.
      </h2>
      <hr />
      <p class="is-size-5" style="padding-bottom: 10px;">
        If you think something is broken, please report it.
      </p>
      <p class="is-size-5">
        Otherwise, head back to the Builds page.
      </p>
      <div class="buttons">
        <a
          class="button has-text-danger is-dark is-uppercase has-text-weight-semibold report-problem"
          style="text-decoration: none;"
          href="https://github.com/joyent/conch-ui/issues/new"
          target="_blank"
        >
          Report a Problem
        </a>
        <a
          class="button is-info is-uppercase has-text-weight-semibold recent-build"
          @click="$router.push({ name: 'builds' })"
        >
          Builds Page
          <span class="icon is-small material-icons" style="margin-left: 10px"
            >arrow_forward</span
          >
        </a>
      </div>
    </div>
    <div class="content-right">
      <img src="../../assets/taken.svg" width="550" />
    </div>
  </div>
</template>
