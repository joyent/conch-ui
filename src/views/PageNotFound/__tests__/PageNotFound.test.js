import PageNotFound from '../PageNotFound.vue';
import { shallowMount } from '@vue/test-utils';

describe('PageNotFound.vue', () => {
  let methods;
  let wrapper;

  beforeEach(() => {
    methods = { routeToMostRecentBuild: jest.fn() };
    wrapper = shallowMount(PageNotFound, { methods });
  });

  test('should render correctly on initial render', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should call routeToMostRecentBuild when "Most Recent Build" button clicked', () => {
    wrapper.find('a.button.recent-build').trigger('click');

    expect(methods.routeToMostRecentBuild).toHaveBeenCalled();
  });
});
