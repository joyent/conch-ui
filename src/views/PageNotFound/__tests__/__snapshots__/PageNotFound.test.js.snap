// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNotFound.vue should render correctly on initial render 1`] = `
<div class="page-not-found hero is-fullheight">
  <div class="content-left">
    <h1 class="title">404</h1>
    <h2 class="subtitle is-5">
      The page you're looking for is... missing.
    </h2>
    <hr>
    <p class="is-size-5">
      Head back to the build you most recently interacted with.
    </p>
    <p class="is-size-5">
      If you think something is broken, please report it.
    </p>
    <div class="buttons"><a class="button is-outlined is-info is-uppercase has-text-weight-semibold recent-build">
        Most Recent Build
      </a> <a href="https://github.com/joyent/conch-ui/issues/new" target="_blank" class="button is-outlined is-danger is-uppercase has-text-weight-semibold report-problem">
        Report a Problem
      </a></div>
  </div>
  <div class="content-right"><img src="../../assets/taken.svg" width="550"></div>
</div>
`;
